src/jit/morph.cpp

```diff
@@ -3388,835 +3170,927 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
-#if defined(_TARGET_X86_)
-            else if (call->IsTailCallViaHelper())
-            {
-                // We have already (before calling fgMorphArgs()) appended the 4 special args
-                // required by the x86 tailcall helper. These args are required to go on the
-                // stack. Force them to the stack here.
-                assert(numArgs >= 4);
-                if (argIndex >= numArgs - 4)
-                {
-                    isRegArg = false;
-                }
-            }
-#endif    // defined(_TARGET_X86_)
-        } // end !reMorphing
+    // First we morph the argument subtrees ('this' pointer, arguments, etc.).
+    // During the first call to fgMorphArgs we also record the
+    // information about late arguments we have in 'fgArgInfo'.
+    // This information is used later to contruct the gtCallLateArgs */
+
+    // Process the 'this' argument value, if present.
+    argx = call->gtCallObjp;
+    if (argx)
+    {
+        fgArgTabEntry* thisArgEntry = call->fgArgInfo->GetArgEntry(0, reMorphing);
+        argx                        = fgMorphTree(argx);
+        call->gtCallObjp            = argx;
+        // This is a register argument - possibly update it in the table.
+        call->fgArgInfo->UpdateRegArg(thisArgEntry, argx, reMorphing);
+        flagsSummary |= argx->gtFlags;
+        assert(argIndex == 0);
+        argIndex++;
+        argSlots++;
+    }
+
+    // Note that this name is a bit of a misnomer - it indicates that there are struct args
+    // that occupy more than a single slot that are passed by value (not necessarily in regs).
+    bool hasMultiregStructArgs = false;
+    for (args = call->gtCallArgs; args; args = args->gtOp.gtOp2, argIndex++)
+    {
+        GenTree**      parentArgx = &args->gtOp.gtOp1;
+        fgArgTabEntry* argEntry   = call->fgArgInfo->GetArgEntry(argIndex, reMorphing);
 
-        //
-        // Now we know if the argument goes in registers or not and how big it is,
-        // whether we had to just compute it or this is a re-morph call and we looked it up.
-        //
-        CLANG_FORMAT_COMMENT_ANCHOR;
+        // Morph the arg node, and update the parent and argEntry pointers.
+        argx        = *parentArgx;
+        argx        = fgMorphTree(argx);
+        *parentArgx = argx;
+        assert(args->OperIsList());
+        assert(argx == args->Current());
 
-#ifdef _TARGET_ARM_
-        // If we ever allocate a floating point argument to the stack, then all
-        // subsequent HFA/float/double arguments go on the stack.
-        if (!isRegArg && passUsingFloatRegs)
+        bool                 passUsingFloatRegs;
+        unsigned             argAlign      = argEntry->alignment;
+        unsigned             size          = argEntry->getSize();
+        CORINFO_CLASS_HANDLE copyBlkClass  = nullptr;
+        bool                 isRegArg      = argEntry->isPassedInRegisters();
+        bool                 isNonStandard = argEntry->isNonStandard;
```

Comment:
There are many unused variables left. For example this `isNonStandard`, `isRegArg` etc.
I think you can get the full list with clang `-Wextra  -Wunused-but-set-parameter `, but maybe it is possible to do somehow else.

Could you please find and delete them?